### PR TITLE
Add `emphasised_organisations` to details hash

### PR DESCRIPTION
**Pull request by jackscotti**
Currently, there are three types of organisations stored in the publishing-api database:

- lead organisations
- organisations
- supporting organisations
Where: organisations = lead organisations + supporting organisations

GOV.UK publishing apps do not send to the publishing-api the same organisation types consistently. We are going to modify all applications to send at least the organisations content ids in the links hash and the "emphasised_organisations" in the details hash. 
Frontend applications will then decide what and how to display the organisations by reading the details and links hashes of a piece of content. 

The `emphasised_organisations` in details hash will allow frontend applications to decide the order in which to display organisations. (emphasised_organisations will go first)

ticket: https://trello.com/c/us3MI1n9/602-fix-organisation-tagging
related to: https://github.com/alphagov/whitehall/pull/2585

Future steps:
- remove lead and supporting organisations from the schemas

### CODE REVIEW

**Comment by boffbowsh** [comment]
I think I prefer `primary_organisations`, or maybe `lead_organisations` should be retained as it matches Whitehall's UI.

**Comment by tijmenb** [comment]
Thanks @boffbowsh. I agree that it's important to keep UI and data naming in sync. 

However in this case it may be better to give it an (overly) specific name, because people could interpret "lead_organisations" as having some sort of real meaning in the context of government.

We also want to use the same mechanism in policy-publisher, where the UI is different:

screen shot 2016-05-12 at 10 19 46

In short, `lead_organisation` could be misinterpreted and lead to people using it in different incompatible ways. Something like `organisations_displayed_first` is easier to change to something more broad, while the reverse is not true. Thoughts?

**Comment by boffbowsh** [comment]
Could just do `organisation_display_order` and list all orgs?

**Comment by fofr** [comment]
> In short, lead_organisation could be misinterpreted and lead to people using it in different incompatible ways. Something like organisations_displayed_first is easier to change to something more broad, while the reverse is not true. Thoughts?

Sounds like we need some user research.

**Comment by MatMoore** [comment]
How about ~~"Boaty McBoatface"~~ "emphasised organisations".

It implies the distinction is purely presentational, but isn't overly specific, and allows us to change the things on the frontend without changing the publishers. I don't think anyone would infer anything about the structure of government from this.

I think the key improvement is moving it to the details hash. The links hash should describe semantics of how things relate to each other, and lead organisation doesn't have a single meaning at the moment. 👍

**Comment by fofr** [comment]
Please try out options with users.

**Comment by boffbowsh** [comment]
This is an internal name @fofr, not for displaying in admin UIs

**Comment by boffbowsh** [comment]
`emphasised_organisations` also works well imo

**Comment by MatMoore** [comment]
Just thinking a little bit more about this -

I'm assuming this means content tagger will have to know about `details`, which it doesn't already? Is this correct? Could we do implement this in such a way that works for all link types?

Regarding user research: I'm aware there has been some research already on publishing workflow, so that may be relevant to helping ensure editors understand the effects of tagging.

I think our (my?) assumptions right now are:
* We're not aware of a user need to order things in an arbitrary way at publishing time.
* The lead/supporting distinction arose because the way we present long lists of organisations looks very odd if you show all of them alphabetically, e.g. the 2012 olympics had lots of organisations tagged to it that people are unlikely to associate with the olympics
* We don't know if there is any end user benefit to this
* We don't have a lot of evidence these organisation links are even useful to end users at all...

**Comment by boffbowsh** [comment]
Content Tagger won't need to know about `details`, only the Publishing apps (mainly Whitehall) would need to know about that in order to add the organisations in the correct order.

**Comment by MatMoore** [comment]
Ah, right forgot organisations is not in content tagger.

**Comment by tijmenb** [comment]
`organisations` _is_ in content-tagger, but setting the `emphasised_organisations` is not its responsibility.

To recap (a consensus):

- We won't change the UIs
- We use `emphasised_organisations` in details hash to allow the frontend to display the things in a certain way
- We will use this pattern for policies (policy-publisher) and whitehall content.

Makes sense?

**Comment by boffbowsh** [comment]
👍  to summary

**Comment by jackscotti** [comment]
👍  thank you all. Updating.

**Comment by fofr** [comment]
> This is an internal name @fofr, not for displaying in admin UIs
Internal names have a habit of making themselves visible, usually for the sake of consistency ;)

**Comment by jackscotti** [comment]
Renamed and added deprecation messages.

**Comment by fofr** [comment]
Should `lead_organisations` still be a required field on some of the formats?
eg https://github.com/alphagov/govuk-content-schemas/blob/master/formats/detailed_guide/publisher/links.json#L6
